<a href="https://colab.research.google.com/github/maryandy2002/maryandy2002/blob/main/MetodoFalsaPosicion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Método de Falsa Posición
A continuación se muestra un código que permite encontrar la raíz entre el intervalo $a=0.5$ y $b=\frac{\pi}{4}$ para la función $f(x)=cos(x)-x$ con una precision de $\varepsilon<10^{-5}$ utilizando el método de falsa posición

In [5]:
import numpy as np
from tabulate import tabulate as tb
import math as mt


en este bloque se importo la librería $tabulate$ que nos permitira realizar una tabla al final de nuestro procedimiento que recopilará los resultados que se fueron obteniendo

In [3]:
def funcion(x):
  return np.cos(x)-x

se define una funcion que será utilizada para ir evaluando valores en $f(x)$ fácilmente

In [28]:
a=[0.5]
b=[mt.pi/4]
n=[1]
c=[0]
e=[]
f=[]

aqui se inicializaron todas las variables que vamos a utilizar, son de tipo lista ya que esto nos facilitara tabular los resultados posteriormente

In [29]:
while (True):
  c.append((a[-1]*funcion(b[-1])-b[-1]*funcion(a[-1]))/(funcion(b[-1])-funcion(a[-1])))
  f.append(funcion(c[-1]))
  if (f[-1]*funcion(a[-1])<0):
    a.append(a[-1])
    b.append(c[-1])
  else:
    b.append(b[-1])
    a.append(c[-1])
  e.append(abs(c[-1]-c[-2]))
  if e[-1]<0.00001:
    break
  n.append(n[-1]+1)
c.pop(0)

0

en este bloque se lleva a cabo todo el procedimiento del método de falsa posición, el cuál, para hasta que el error $\varepsilon$ sea menor que $0.00001$, mientras esto no ocurra se va tomando el valor de $c$ en el que la recta que une a los puntos $(a,f(a))$ y $(b,f(b))$ interseca con el eje $x$, es decir cuando $y=0$ y dependiendo del signo de la funcion evaluada
$f(c)$ se renueva el nuevo intervalo para poder obtener un resultado cada vez mas preciso

In [30]:
table = [n,a,b,c,f,e]
print(tb((zip(*table)),headers=["n","a","b","c","f(c)","e"]))

  n         a         b         c         f(c)            e
---  --------  --------  --------  -----------  -----------
  1  0.5       0.785398  0.736384  0.00451772   0.736384
  2  0.736384  0.785398  0.739058  4.51772e-05  0.002674
  3  0.739058  0.785398  0.739085  4.50872e-07  2.67246e-05
  4  0.739085  0.785398  0.739085  4.49964e-09  2.66712e-07


finalmente aqui se tabularon nuestros resultados, esto nos permiten visualizar con facilidad como fue cambiando el valor de nuestra aproximacion a la raíz el cual es el ultimo dato en la columna $c$

Éstos resultados nos permiten concluir que para la funcion $f(x)=cos(x)-x$ con un $\varepsilon<10^{-5}$ es $c=0.739085$